src/microbit/model/button.py

```diff
@@ -0,0 +1,26 @@
+class Button:
+    def __init__(self):
+        self.__pressed = False
+        self.__presses = 0
+        self.__prev_pressed = False
+
+    def is_pressed(self):
+        return self.__pressed
+
+    def was_pressed(self):
+        res = self.__prev_pressed
+        self.__prev_pressed = False
+        return res
+
+    def get_presses(self):
+        res = self.__presses
+        self.__presses = 0
+        return res
+
+    def __press_down(self):
```

Comment:
The front end side will be calling these functions, and so it can register clicks and releases.
If someone is holding it, then I wouldn't want it to call release right away. When that person releases their button click, it will call the release function. Theoretically, __press_down should not be called twice without a release between them. Let me know what you think or if there might be any problems with that logic.

Reply:
Up to you, just double checking whether we are considering these cases.

Reply:
Def considered! Will be aware of it and thinking about the cases for the integration.